generator/testing/fake_source_tree.cc

```diff
@@ -26,8 +26,8 @@ void FakeSourceTree::Insert(std::string const& filename, std::string contents) {
 }
 
 google::protobuf::io::ZeroCopyInputStream* FakeSourceTree::Open(
-    std::string const& filename) {
-  auto iter = files_.find(filename);
+    FilenameType filename) {
+  auto iter = files_.find(std::string(filename));
```

Comment:
Observation:  It's a shame we make a copy (particularly in a "zero copy" function :-), but no biggie.

Reply:
Indeed.